nixos/modules/programs/wayland/hyprland.nix

```diff
@@ -57,24 +64,23 @@ in
     nvidiaPatches = mkEnableOption (mdDoc "patching wlroots for better Nvidia support");
   };
 
-  config = mkIf cfg.enable {
-    environment.systemPackages = [ cfg.finalPackage ];
+  config = mkIf cfg.enable (mkMerge [
+    {
+      environment.systemPackages = optional (cfg.finalPackage != null) cfg.finalPackage;
 
-    fonts.enableDefaultPackages = mkDefault true;
-    hardware.opengl.enable = mkDefault true;
+      # To make a Hyprland session available if a display manager like SDDM is enabled:
+      services.xserver.displayManager.sessionPackages = optional (cfg.finalPackage != null) cfg.finalPackage;
 
-    programs = {
-      dconf.enable = mkDefault true;
-      xwayland.enable = mkDefault cfg.xwayland.enable;
-    };
-
-    security.polkit.enable = true;
-
-    services.xserver.displayManager.sessionPackages = [ cfg.finalPackage ];
+      # For screen sharing
+      xdg.portal = {
```

Comment:
Shouldn't xdg portals be enabled inside `wayland-session.nix`? They are an important part of the Wayland desktop experience and should be enabled by default.

Reply:
I moved this from `wayland-session.nix` into each compositor's module after [this](https://github.com/NixOS/nixpkgs/pull/240989#discussion_r1271114038) comment.

Reply:
Yes, for `extraPortals` that makes sense. But `xdg.portal.enable = true;` should be enabled everywhere.

Reply:
But isn't it useless without any portal package?

Reply:
We seem to only have wlroots compositors in scope. So `xdg.portals.wlr.enable` can be set safely, even for Hyprland. The hypr portal will be automatically picked anyway.